pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -710,7 +710,26 @@ def infer_model(model: Model | KnownModelName | str) -> Model:  # noqa: C901
         )
         provider = 'google-vertex'
 
-    if provider == 'cohere':
+    if provider == 'gateway':
+        from ..providers.gateway import GatewayProvider
+
+        # TODO(Marcelo): We need to try..except ValueError.
+        forward_provider, model_name = model_name.split('/', 1)
+        if forward_provider in ('openai', 'openai-chat'):
```

Comment:
should this be a match statement? Should we move it into a new method within the gateway code? Presumably we'll support many more models.